others/iterative_tree_traversals.cpp

```diff
@@ -0,0 +1,362 @@
+    std::vector<int64_t> actual_result{4, 5, 2, 3, 1};
+    std::vector<int64_t> result; ///< result stores the postorder traversal of the binary tree.
+
+    // Calling postOrderIterative() function by passing a root node,
+    // and storing the postorder traversal in result.
+    result = binaryTree.postOrderIterative(root);
+
+    // Self-testing the result using `assert`
+    for(int i = 0; i < result.size(); i++)
+        assert(actual_result[i] == result[i]);
+
+    // Printing the result storing postorder.
+    std::cout<< "\nPostOrder Traversal Is : "<< std::endl;
+    for(auto i: result) {
+        std::cout<< i<< "  ";
+    }
+}
+
+/**
+ * @brief Test the computed inorder with the actual inorder.
+ * @param binaryTree instance of BinaryTree class
+ * @param root head/root node of a tree
+ */
+static void test3(others::iterative_tree_traversals::BinaryTree binaryTree, others::iterative_tree_traversals::Node *root){
+    std::vector<int64_t> actual_result{4, 2, 5, 1, 3};
+    std::vector<int64_t> result; ///< result stores the inorder traversal of the binary tree.
+
+    // Calling inOrderIterative() function by passing a root node,
+    // and storing the inorder traversal in result.
+    result = binaryTree.inOrderIterative(root);
+
+    // Self-testing the result using `assert`
+    for(int i = 0; i < result.size(); i++)
+        assert(actual_result[i] == result[i]);
+
+    // Printing the result storing inorder.
+    std::cout<< "\nInOrder Traversal Is : "<< std::endl;
+    for(auto i: result) {
+        std::cout<< i<< "  ";
+    }
+}
+
+/**
+ * @brief Test the computed preorder with the actual preorder on negative value.
+ * @param binaryTree instance of BinaryTree class
+ * @param root head/root node of a tree
+ */
+static void test4(others::iterative_tree_traversals::BinaryTree binaryTree, others::iterative_tree_traversals::Node *root){
+    std::vector<int64_t> actual_result{-1, -2, -4, -5, -3};
+    std::vector<int64_t> result; ///< result stores the preorder traversal of the binary tree
+
+    // Calling preOrderIterative() function by passing a root node,
+    // and storing the preorder traversal in result.
+    result = binaryTree.preOrderIterative(root); 
+
+    // Self-testing the result using `assert`
+    for(int i = 0; i < result.size(); i++)
+        assert(actual_result[i] == result[i]);
+
+    // Printing the result storing preorder.
+    std::cout<< "\nPreOrder Traversal Is : "<< std::endl;
+    for(auto i: result) {
+        std::cout<< i<< "  ";
+    }
+}
+
+/**
+ * @brief Test the computed postorder with the actual postorder on negative value.
+ * @param binaryTree instance of BinaryTree class
+ * @param root head/root node of a tree
+ */
+static void test5(others::iterative_tree_traversals::BinaryTree binaryTree, others::iterative_tree_traversals::Node *root){
+    std::vector<int64_t> actual_result{-4, -5, -2, -3, -1};

```

Comment:
Please add a one-line description of what the function is for (like the line below).